Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/CancelRide.hs

```diff
@@ -86,22 +102,25 @@ cancelRideHandler ServiceHandle {..} requestorId rideId req = withLogTag ("rideI
           driver <- findById driverId >>= fromMaybeM (PersonNotFound driverId.getId)
           unless (authPerson.merchantId == driver.merchantId) $ throwError (RideDoesNotExist rideId.getId)
           logTagInfo "admin -> cancelRide : " ("DriverId " <> getId driverId <> ", RideId " <> getId ride.id)
-          buildRideCancelationReason Nothing DBCR.ByMerchant ride
+          buildRideCancelationReason Nothing Nothing Nothing Nothing DBCR.ByMerchant ride
         DP.DRIVER -> do
           unless (authPerson.id == driverId) $ throwError NotAnExecutor
           logTagInfo "driver -> cancelRide : " ("DriverId " <> getId driverId <> ", RideId " <> getId ride.id)
-          buildRideCancelationReason (Just driverId) DBCR.ByDriver ride
+          location <- findDriverLocationId driverId >>= fromMaybeM LocationNotFound
```

Comment:
If driver location doesn't find, don't throw error, just update it as Nothing. As cancellation should not fail if we don't have driver location.

Reply:
pushed new changes.